operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/CustomResourceCache.java

```diff
@@ -3,42 +3,39 @@
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import io.fabric8.kubernetes.client.CustomResource;
-import java.util.Map;
 import java.util.Optional;
 import java.util.concurrent.ConcurrentHashMap;
-import java.util.concurrent.locks.Lock;
-import java.util.concurrent.locks.ReentrantLock;
+import java.util.concurrent.ConcurrentMap;
 import java.util.function.Predicate;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+@SuppressWarnings("rawtypes")
 public class CustomResourceCache {
 
   private static final Logger log = LoggerFactory.getLogger(CustomResourceCache.class);
 
-  private ObjectMapper objectMapper = new ObjectMapper();
-  private final Map<String, CustomResource> resources = new ConcurrentHashMap<>();
-  private final Lock lock = new ReentrantLock();
+  private final ObjectMapper objectMapper;
+  private final ConcurrentMap<String, CustomResource> resources = new ConcurrentHashMap<>();
+
+  public CustomResourceCache(ObjectMapper objectMapper) {
```

Comment:
I would add a default no-arg constructor that would just use the default ObjectMapper. This way, existing code wouldn't have to be changed.

Reply:
Sure, admittedly it is a public class though I don't see why you would use it outside of the library itself.

Reply:
This would allow to leave the test classes untouched, for example… 😉